tests: tests/net/lib/http_server: build failure

**Describe the bug**

<html>
<body>

CASE_NAME | ZEPHYR_VERSION | RESULT | PLATFORM | TEST_PATH
-- | -- | -- | -- | --
net.http.server.common | v3.7.0-485-gebd31d3916ab | Build failure | ls1046ardb | tests/net/lib/http_server/common
net.http.server.core | v3.7.0-485-gebd31d3916ab | Build failure | ls1046ardb | tests/net/lib/http_server/core
net.http.server.crime | v3.7.0-485-gebd31d3916ab | Build failure | ls1046ardb | tests/net/lib/http_server/crime
net.http.server.common | v3.7.0-485-gebd31d3916ab | Build failure | ls1046ardb/ls1046a/smp/4cores | tests/net/lib/http_server/common
-- | -- | -- | -- | --
net.http.server.core | v3.7.0-485-gebd31d3916ab | Build failure | ls1046ardb/ls1046a/smp/4cores | tests/net/lib/http_server/core
net.http.server.crime | v3.7.0-485-gebd31d3916ab | Build failure | ls1046ardb/ls1046a/smp/4cores | tests/net/lib/http_server/crime
net.http.server.crime | v3.7.0-485-gebd31d3916ab | Build failure | ls1046ardb/ls1046a/smp | tests/net/lib/http_server/crime
-- | -- | -- | -- | --
net.http.server.common | v3.7.0-485-gebd31d3916ab | Build failure | ls1046ardb/ls1046a/smp | tests/net/lib/http_server/common
net.http.server.core | v3.7.0-485-gebd31d3916ab | Build failure | ls1046ardb/ls1046a/smp | tests/net/lib/http_server/core
</body>
</html>


Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - ls1046ardb
 - What have you tried to diagnose or workaround this issue?
git bisect 
- Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 
```
e55278a87be4e5534ba772b23e459882866bcd6d is the first bad commit
commit e55278a87be4e5534ba772b23e459882866bcd6d
Author: Gerhard Jörges <joerges@metratec.com>
Date:   Fri Jul 26 10:54:42 2024 +0200

    net: lib: http_server: add static fs resource
    
    adds filesystem as a resource for the http_server which serves static
    (gzipped) files from a filesystem to the client.
    
    Signed-off-by: Gerhard Jörges <joerges@metratec.com>

 doc/connectivity/networking/api/http_server.rst |  32 +++++++
 include/zephyr/net/http/server.h                |  34 +++++++
 subsys/net/lib/http/CMakeLists.txt              |   4 +
 subsys/net/lib/http/headers/server_internal.h   |   3 +
 subsys/net/lib/http/http_server_core.c          |  47 +++++++++-
 subsys/net/lib/http/http_server_http1.c         | 120 ++++++++++++++++++++++--
 subsys/net/lib/http/http_server_http2.c         | 108 +++++++++++++++++++++
 subsys/net/lib/http/iterables.ld                |   9 ++
 subsys/net/lib/shell/http.c                     |   3 +
 9 files changed, 351 insertions(+), 9 deletions(-)
 create mode 100644 subsys/net/lib/http/iterables.ld

```


**To Reproduce**

west twister -p ls1046ardb  -T tests/net/lib/http_server/common --build-only

**Expected behavior**
test PASS

**Impact**
<!--
What impact does this issue have on your progress (e.g., annoyance, showstopper)
-->

**Logs and console output**

```
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/net/lib/http/http_server_http1.c
In file included from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/logging/log.h:11,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/net/lib/http/http_server_http1.c:20:
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/net/lib/http/http_server_http1.c: In function 'handle_http1_static_fs_resource':
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/net/lib/http/http_server_http1.c:333:17: error: format '%d' expects argument of type 'int', but argument 3 has type 'size_t' {aka 'long unsigned int'} [-Werror=format=]
  333 |         LOG_DBG("found %s, file size: %d", fname, file_size);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~         ~~~~~~~~~
      |                                                   |
      |                                                   size_t {aka long unsigned int}
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/logging/log_core.h:275:42: note: in definition of macro 'Z_LOG2'
  275 |                 z_log_printf_arg_checker(__VA_ARGS__); \
      |                                          ^~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/logging/log.h:75:25: note: in expansion of macro 'Z_LOG'
   75 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/net/lib/http/http_server_http1.c:333:9: note: in expansion of macro 'LOG_DBG'
  333 |         LOG_DBG("found %s, file size: %d", fname, file_size);
      |         ^~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/net/lib/http/http_server_http1.c:333:40: note: format string is defined here
  333 |         LOG_DBG("found %s, file size: %d", fname, file_size);
      |                                       ~^
      |                                        |
      |                                        int
      |                                       %ld
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.

INFO    - /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/ls1046ardb/tests/net/lib/http_server/core/net.http.server.core/build.log
```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v3.7.0-485-gebd31d3916ab

--- a/subsys/net/lib/http/http_server_http1.c
+++ b/subsys/net/lib/http/http_server_http1.c
@@ -330,7 +330,7 @@ int handle_http1_static_fs_resource(struct http_resource_detail_static_fs *stati
 		}
 	}
 
-	LOG_DBG("found %s, file size: %d", fname, file_size);
+	LOG_DBG("found %s, file size: %zu", fname, file_size);
 
 	/* send HTTP header */
 	len = snprintk(http_response, sizeof(http_response), RESPONSE_TEMPLATE_STATIC_FS,

